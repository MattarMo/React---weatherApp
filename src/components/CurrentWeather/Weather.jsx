/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import EnterLocationWeather from './EnterLocationWeather.jsx';
import DisplayWeather from './DisplayWeather.jsx';
import { connect } from 'react-redux';
import api from '../../api';
import * as actionTypes from '../../store/actions';

class CurrentWeather extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: 'Get your Current Weather for today'
    };
  }

  render() {
    return (
      <div>
        <EnterLocationWeather
          title={this.state.title}
          cityvalue={this.props.searchCity}
          countryvalue={this.props.searchCountry}
          inputchangecity={this.props.handleInputCityChange}
          inputchangecountry={this.props.handleInputCountryChange}
          getWeather={this.props.getWeather}
        />
        <DisplayWeather
          temperature={this.props.temperature}
          description={this.props.description}
          icon={this.props.icon}
          error={this.props.error}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    temperature: undefined,
    description: undefined,
    icon: undefined,
    error: state.error,
    searchCity: state.searchForecastCityInput,
    searchCountry: state.searchForecastCountryInput
  };
};

const mapDispatchToProps = dispatch => {
  return {
    handleInputCityChange: e => {
      dispatch({
        type: actionTypes.SEARCH_WEATHERCITY_INPUT,
        cityvalue: e.target.value
      });
    },
    handleInputCountryChange: e => {
      dispatch({
        type: actionTypes.SEARCH_WEATHERCOUNTRY_INPUT,
        countryvalue: e.target.value
      });
    },
    getWeather: (e, city, country) => {
      e.preventDefault();
      api.getWeather(dispatch, city, country);
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CurrentWeather);
