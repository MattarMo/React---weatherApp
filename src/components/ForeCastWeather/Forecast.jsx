/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import EnterLocationForecast from './EnterLocationForecast.jsx';
import DisplayForecast from './DisplayForecast.jsx';
import { connect } from 'react-redux';
import api from '../../api';
import * as actionTypes from '../../store/actions';

class Forecast extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: 'Get your 5 day Forecast'
    };
  }

  render() {
    return (
      <div>
        <EnterLocationForecast
          title={this.state.title}
          cityvalue={this.props.searchCity}
          countryvalue={this.props.searchCountry}
          inputchangecity={this.props.handleInputCityChange}
          inputchangecountry={this.props.handleInputCountryChange}
          getForecast={this.props.getForeCast}
        />
        <DisplayForecast
          forecastData={this.props.forecastDataList}
          error={this.props.error}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    forecastDataList: state.forecastData,
    error: state.error,
    searchCity: state.searchForecastCityInput,
    searchCountry: state.searchForecastCountryInput
  };
};

const mapDispatchToProps = dispatch => {
  return {
    handleInputCityChange: e => {
      dispatch({
        type: actionTypes.SEARCH_FORECASTCITY_INPUT,
        cityvalue: e.target.value
      });
    },
    handleInputCountryChange: e => {
      dispatch({
        type: actionTypes.SEARCH_FORECASTCOUNTRY_INPUT,
        countryvalue: e.target.value
      });
    },
    getForeCast: (e, city, country) => {
      e.preventDefault();
      api.getForecast(dispatch, city, country);
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Forecast);
