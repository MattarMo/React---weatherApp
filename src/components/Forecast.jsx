import React, { Component } from 'react';

export default class Forcast extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit(e) {
    console.log(
      'UserName: \n' +
        this.state.username +
        '\nPassWord:\n' +
        this.state.password
    );
    e.preventDefault();
  }
  render() {
    return (
      <div className="row">
        <div className="col-md-3 align-self-center" />
        <div className="card text-white border-primary mb-3 text-center">
          <div className="card-header bg-light text-dark">Login Form</div>
          <form className="align-content-center">
            <div className="card-body bg-secondary">
              <div className="form-group">
                <label htmlFor="username">Username</label>
                <br />
                <input
                  className="form-control"
                  type="text"
                  name="username"
                  value={this.state.loginInfo}
                  onChange={this.handleChange}
                />
              </div>
              <div className="form-group">
                <label htmlFor="password">Password</label>
                <br />
                <input
                  className="form-control"
                  type="text"
                  name="password"
                  value={this.state.loginInfo}
                  onChange={this.handleChange}
                />
              </div>
            </div>
            <div className="card-footer text-muted">
              <button type="submit" className="btn btn-group-lg btn-primary">
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
