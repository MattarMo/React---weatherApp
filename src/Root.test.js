import React from 'react';
import renderer from 'react-test-renderer';
import Forecast from './Components/ForeCastWeather/Forecast.jsx';
import Weather from './Components/CurrentWeather/Weather.jsx';

describe('Forecast.Weather Components', () => {
  it('Forecast component matches snapshot', () => {
    const tree = renderer.create(<Forecast />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Weather component matches snapshot', () => {
    const tree = renderer.create(<Weather />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
