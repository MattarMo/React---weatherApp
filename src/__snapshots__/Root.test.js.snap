// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Component Forecast component matches snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <h2
      className="align-text-top"
    >
      Get your 5 day Forecast
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="City"
        >
          City
        </label>
        <input
          className="form-control"
          name="city"
          placeholder="Enter your city"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="country"
        >
          Country
        </label>
        <input
          className="form-control"
          name="country"
          placeholder="Enter your country"
          type="text"
        />
      </div>
      <button
        className="btn btn-primary btn-"
      >
        Get Current Weather
      </button>
    </form>
  </div>
  <div
    className="container"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Date/Time
          </th>
          <th
            scope="col"
          >
            Temperature
          </th>
          <th
            scope="col"
          >
            WeatherIcon
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Forecast Component Weather component matches snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <h2>
      Get your Current Weather for today
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="City"
        >
          City
        </label>
        <input
          className="form-control"
          name="city"
          placeholder="Enter your city"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="country"
        >
          Country
        </label>
        <input
          className="form-control"
          name="country"
          placeholder="Enter your country"
          type="text"
        />
      </div>
      <button
        className="btn btn-primary"
      >
        Get Current Weather
      </button>
    </form>
  </div>
  <div
    className="container"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Temperature
          </th>
          <th
            scope="col"
          >
            Description
          </th>
          <th
            scope="col"
          >
            WeatherIcon
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="table"
        />
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Forecast Component matches snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <h2
      className="align-text-top"
    >
      Get your 5 day Forecast
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="City"
        >
          City
        </label>
        <input
          className="form-control"
          name="city"
          placeholder="Enter your city"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="country"
        >
          Country
        </label>
        <input
          className="form-control"
          name="country"
          placeholder="Enter your country"
          type="text"
        />
      </div>
      <button
        className="btn btn-primary btn-"
      >
        Get Current Weather
      </button>
    </form>
  </div>
  <div
    className="container"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Date/Time
          </th>
          <th
            scope="col"
          >
            Temperature
          </th>
          <th
            scope="col"
          >
            WeatherIcon
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Forecast.Weather Components Forecast component matches snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <h2
      className="align-text-top"
    >
      Get your 5 day Forecast
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="City"
        >
          City
        </label>
        <input
          className="form-control"
          name="city"
          placeholder="Enter your city"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="country"
        >
          Country
        </label>
        <input
          className="form-control"
          name="country"
          placeholder="Enter your country"
          type="text"
        />
      </div>
      <button
        className="btn btn-primary btn-"
      >
        Get Current Weather
      </button>
    </form>
  </div>
  <div
    className="container"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Date/Time
          </th>
          <th
            scope="col"
          >
            Temperature
          </th>
          <th
            scope="col"
          >
            WeatherIcon
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Forecast.Weather Components Weather component matches snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <h2>
      Get your Current Weather for today
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="City"
        >
          City
        </label>
        <input
          className="form-control"
          name="city"
          placeholder="Enter your city"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="country"
        >
          Country
        </label>
        <input
          className="form-control"
          name="country"
          placeholder="Enter your country"
          type="text"
        />
      </div>
      <button
        className="btn btn-primary"
      >
        Get Current Weather
      </button>
    </form>
  </div>
  <div
    className="container"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Temperature
          </th>
          <th
            scope="col"
          >
            Description
          </th>
          <th
            scope="col"
          >
            WeatherIcon
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="table"
        />
      </tbody>
    </table>
  </div>
</div>
`;
